.github/workflows/full-ci-cd.yaml

```diff
@@ -4,6 +4,10 @@ on:
   push:
     branches: [main]
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
```

Comment:
I think `github.ref` references the commit SHA, right? So the group will change on every merge. 

From the docs:

> The concurrency key is used to group workflows or jobs together into a concurrency group. When you define a concurrency key, GitHub Actions ensures that only one workflow or job with that key runs at any given time. If a new workflow run or job starts with the same concurrency key, GitHub Actions will cancel any workflow or job already running with that key. The concurrency key can be a hard-coded string, or it can be a dynamic expression that includes context variables.

I think for our case the group should be hardcoded, e.g. continuous-deployment.